kernel-spark/src/test/java/io/delta/kernel/spark/read/SparkGoldenTableTest.java

```diff
@@ -563,16 +544,8 @@ public void testPartitionedTable() {
     }
     Dataset<Row> df = full.selectExpr(projectedCols.toArray(new String[0]));
 
-    Function0<Dataset<Row>> dfFunc =
-        new Function0<Dataset<Row>>() {
-          @Override
-          public Dataset<Row> apply() {
-            return df;
-          }
-        };
-    scala.collection.immutable.Seq<Row> expectedSeq =
-        scala.collection.JavaConverters.asScalaBuffer(expected).toList();
-    checkAnswer(dfFunc, expectedSeq);
```

Comment:
we should use the checkAnswer instead of simply verifying the length of dataframes